src/main/java/org/apache/mesos/hdfs/executor/AbstractNodeExecutor.java

```diff
@@ -125,10 +127,28 @@ private void createSymbolicLink() {
       Files.createSymbolicLink(hdfsLinkDirPath, sandboxHdfsBinaryPath);
       log.info("The linked HDFS binary path is: " + sandboxHdfsBinaryPath);
       log.info("The symbolic link path is: " + hdfsLinkDirPath);
+      // Adding binary to the PATH environment variable
+      addBinaryToPath(hdfsBinaryPath);
     } catch (Exception e) {
-      log.error("Error creating the symbolic link to hdfs binary: " + e);
+      log.fatal("Error creating the symbolic link to hdfs binary: " + e);
```

Comment:
Nope, it recreates the files each time so the latest binary is distributed and linked.